lua/modules/controls.lua

```diff
@@ -1181,16 +1181,21 @@ _state.upListener = LocalEvent:Listen(LocalEvent.Name.PointerUp, function(pointe
 			local indicator = _getOrCreateIndicator(pointerEvent.Index)
 			indicator:_hide()
 
+			local i1 = _state.touchZoomAndDrag2Pointers[1].index
+			local i2 = _state.touchZoomAndDrag2Pointers[2].index
+			local wasDragPointer = _pointerIndexWithin(pointerEvent.Index, touchDragPointer, i1, i2)
+
+			if wasDragPointer and Pointer.Up then
+				Pointer.Up(pointerEvent)
+			end
+
 			if pointerEvent.Index == touchDragPointer then
 				if _state.clickPointerIndex ~= nil and pointerEvent.Index == _state.clickPointerIndex then
 					if Pointer.Click ~= nil then
 						Pointer.Click(pointerEvent)
 					end
 					_state.clickPointerIndex = nil
 				end
-				if Pointer.Up ~= nil then
```

Comment:
@XLegland I don't understand why that is necessary?
`Up` has to happen before `Click`, at least this change is breaking that assumption. 

Also `Pointer.Click` and `Pointer.Up` are necessarily triggered by the same pointer event.